### ISSUE TITLE
rsync__args being ignored

### ISSUE THREAD

**Issue by 2bithacker** [hashicorp/vagrant#9604]
### Vagrant version
Vagrant 2.0.2

### Host operating system
Linux Mint 18.3 Sylvia

### Guest operating system
Ubuntu 14.04.5 LTS

### Vagrantfile
```ruby
Vagrant.configure("2") do |config|
  config.vm.define "miticon" do |config|
    config.vm.provider :digital_ocean do |provider, override|
      override.ssh.private_key_path = '~/.ssh/id_vagrant'
      override.vm.box = 'digital_ocean'
      override.vm.box_url = "https://github.com/devopsgroup-io/vagrant-digitalocean/raw/master/box/digital_ocean.box"
      override.nfs.functional = false
      provider.token = 'TOKEN-REMOVED'
      provider.image = 'ubuntu-14-04-x64'
      provider.region = 'tor1'
      provider.size = 's-1vcpu-1gb'
    end
    config.vm.synced_folder ".", "/vagrant", type: "rsync", 
      rsync__args: ["-avz", "--copy-links"],
      rsync__exclude: [".git/","venv"]
  end
end
```

### Debug output
https://gist.github.com/2bithacker/64c826049681848488908ca361d5e961

### Expected behavior
`rsync` should be run with the arguments specified in the `rsync__args` parameter.

### Actual behavior
`rsync` was run with the default arguments.

### Steps to reproduce
1. `vagrant rsync`

### References
- #5911 - identical issue, also opened with the digitalocean plugin's repo as devopsgroup-io/vagrant-digitalocean#202, which says it was fixed, however vagrant-digitalocean's rsync code was removed in devopsgroup-io/vagrant-digitalocean#222

**Comment by 2bithacker** [comment]
Using vagrant-digitalocean (0.9.2) by the way.

**Comment by briancain** [comment]
Hello @2bithacker - unfortunately, this is an issue with the Digital Ocean provider you are using. Here is a section of the debug log where the rsync takes place

```
 INFO guest: Execute capability: rsync_pre [#<Vagrant::Machine: miticon (VagrantPlugins::DigitalOcean::Provider)>, {:type=>"rsync", :rsync__exclude=>[".git/", "venv"], :guestpath=>"/vagrant", :hostpath=>"/home/cmarshall/Work/miticon", :disabled=>false, :__vagrantfile=>true, :exclude=>[".git/", "venv"], :owner=>"root", :group=>"root"}] (ubuntu)
DEBUG ssh: Re-using SSH connection.
 INFO ssh: Execute: mkdir -p /vagrant (sudo=true)
DEBUG ssh: stderr: stdin: is not a tty

DEBUG ssh: stderr: 41e57d38-b4f7-4e46-9c38-13873d338b86-vagrant-ssh
DEBUG ssh: Exit status: 0
 INFO subprocess: Starting process: ["/usr/bin/rsync", "--verbose", "--archive", "--delete", "-z", "--copy-links", "--no-owner", "--no-group", "--rsync-path", "sudo rsync", "-e", "ssh -p 22 -o LogLevel=FATAL  -o ControlMaster=auto -o ControlPath=/tmp/ssh.923 -o ControlPersist=10m  -o IdentitiesOnly=yes -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i '/home/cmarshall/.ssh/id_vagrant'", "--exclude", ".vagrant/", "--exclude", ".git/", "--exclude", "venv", "/home/cmarshall/Work/miticon/", "root@138.197.174.151:/vagrant"]
 INFO subprocess: Command not in installer, restoring original environment...
```

If you notice, it's saying that it's using the digital oceans `rsync_pre` capability. Specifically:

 ```ruby
rsync_pre [#<Vagrant::Machine: miticon (VagrantPlugins::DigitalOcean::Provider)
```

And also it shows the rsync command that's being run, which does ignore your args. So it looks like this is a problem with their provider.

For contrast, I tried this with the VirtualBox provider, and it works as expected:

```
INFO guest: Execute capability: rsync_pre [#<Vagrant::Machine: bork (VagrantPlugins::ProviderVirtualBox::Provider)>, {:type=>:rsync, :rsync__args=>["-avz", "--copy-links"], :guestpath=>"/vagrant", :hostpath=>"/Users/brian/code/vagrant-sandbox/scripts", :disabled=>false, :__vagrantfile=>true, :args=>["-avz", "--copy-links"], :owner=>"vagrant", :group=>"vagrant"}] (ubuntu)
DEBUG ssh: Re-using SSH connection.
 INFO ssh: Execute: mkdir -p /vagrant (sudo=true)
DEBUG ssh: stderr: mesg:
DEBUG ssh: stderr: ttyname failed
DEBUG ssh: stderr: :
DEBUG ssh: stderr: Inappropriate ioctl for device

DEBUG ssh: stderr: 41e57d38-b4f7-4e46-9c38-13873d338b86-vagrant-ssh
DEBUG ssh: Exit status: 0
 INFO subprocess: Starting process: ["/usr/bin/rsync", "-avz", "--copy-links", "--no-owner", "--no-group", "--rsync-path", "sudo rsync", "-e", "ssh -p 2222 -o LogLevel=FATAL  -o ControlMaster=auto -o ControlPath=/var/folders/nf/pw544nm16fd9gf33kp04047r0000gn/T/ssh.928 -o ControlPersist=10m  -o IdentitiesOnly=yes -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i '/Users/brian/code/vagrant-sandbox/.vagrant/machines/bork/virtualbox/private_key'", "--exclude", ".vagrant/", "/Users/brian/code/vagrant-sandbox/scripts/", "vagrant@127.0.0.1:/vagrant"]
```

I recommend opening an issue with them, and feel free to redirect them to this issue in case they think it's a problem with Vagrant. Thanks! 👍

**Comment by briancain** [comment]
I also recommend upgrading to the latest Vagrant and see if that fixes anything....I only tried Vagrant 2.1.4. Perhaps it's been fixed since 2.0.2?

**Comment by briancain** [comment]
Your debug log also does say it's using Vagrants default `linux` capability rsync_pre, so I wonder if it's just a matter of upgrading to the latest version:

```
DEBUG guest: Found cap: rsync_pre in linux
```
